proto/user-service.proto

```diff
@@ -52,6 +64,20 @@ message UpdateUserResponse {
     User user = 1;
 }
 
+message ValidateTokenResponse {
+    bool valid = 1;
+}
+
+message LoginRequest {
+    string username = 1;
```

Comment:
But isn't that check (token.username == request.username) really not necessary?

If the token has a claim for username "superawesomeuser", the token belongs to "superawesomeuser" and anything we're doing that requires a token should use the "superawesomeuser" user.
The only way you should be able to get a token is by sending a request to the "/login" with username and password. That token is never given away any other way.

Also the username in the token can easily be retrieved from the token. Simply decode the message from base64 into ascii (or whatever).  So that check doesn't really add any security